This patch was prepared by Gwenole Beauchesne <gbeauchesne@mandrakesoft.com> for Mandrake
# Patch22: Improve font server discovery (OOo 1.0-branch)
--- oo_1.0_src/psprint/source/fontmanager/fontmanager.cxx.psprint-fontmanager	2002-02-28 12:49:51.000000000 +0100
+++ oo_1.0_src/psprint/source/fontmanager/fontmanager.cxx	2002-05-28 11:20:32.000000000 +0200
@@ -1602,14 +1602,13 @@ void PrintFontManager::initialize( void*
     char** pPaths = XGetFontPath( pDisplay, &nPaths );
 
     int nPos = 0;
-    int nPort = 0;
     bool bServerDirs = false;
     for( i = 0; i < nPaths; i++ )
     {
         ByteString aPath( pPaths[i] );
         if( ! bServerDirs
             && ( nPos = aPath.Search( ':' ) ) != STRING_NOTFOUND
-            && ( nPort = aPath.Copy( nPos+1 ).ToInt32() ) > 0 )
+            && ( !aPath.Equals( ":unscaled", nPos, 9 ) ) )
         {
             bServerDirs = true;
             getServerDirectories();
